core/src/main/java/org/apache/brooklyn/core/BrooklynFeatureEnablement.java

```diff
@@ -182,27 +211,38 @@ public static boolean setEnablement(String property, boolean val) {
         synchronized (MUTEX) {
             boolean oldVal = isEnabled(property);
             FEATURE_ENABLEMENTS.put(property, val);
+            
+            if (val == oldVal) {
+                LOG.debug("Enablement of "+property+" set to explicit "+val+" (no-op as resolved to same value previously)");
+            } else {
+                LOG.debug("Enablement of "+property+" set to explicit "+val+" (previously resolved to "+oldVal+")");
+            }
             return oldVal;
         }
     }
     
     public static void setDefault(String property, boolean val) {
         synchronized (MUTEX) {
-            if (!FEATURE_ENABLEMENTS.containsKey(property)) {
-                String rawVal = System.getProperty(property);
-                if (rawVal == null) {
-                    FEATURE_ENABLEMENTS.put(property, val);
-                    LOG.debug("Default enablement of "+property+" set to "+val);
+            Boolean oldDefaultVal = FEATURE_ENABLEMENT_DEFAULTS.get(property);
+            FEATURE_ENABLEMENT_DEFAULTS.put(property, val);
+            if (oldDefaultVal != null) {
+                if (oldDefaultVal.equals(val)) {
+                    LOG.debug("Default enablement of "+property+" set to "+val+" (no-op as same as previous default value)");
                 } else {
-                    LOG.debug("Not setting default enablement of "+property+" to "+val+", because system property is "+rawVal);
+                    LOG.debug("Default enablement of "+property+" set to "+val+" (overwriting previous default of "+oldDefaultVal+")");
                 }
+            } else {
+                LOG.debug("Default enablement of "+property+" set to "+val);
             }
         }
     }
-    
+
+    @VisibleForTesting
     static void clearCache() {
         synchronized (MUTEX) {
             FEATURE_ENABLEMENTS.clear();
```

Comment:
Unrelated to this PR, but if `clearCache` is called, there is no mechanism for downstream projects to re-add their defaults